CacheService/docker_context/Client.py

```diff
@@ -0,0 +1,87 @@
+import asyncio
+
+import aiohttp
+from aiohttp import ClientResponseError
+from sanic.log import logger
+
+
+class Client:
+
+    SERVICE_UNAVAILABLE_STATUS = 503
+
+    def __init__(self, loop):
+        self.client = aiohttp.ClientSession(loop=loop)
+        self.loop = loop
+
+    async def getJson(self, response):
+        result = None
+        try:
+            result = await response.json()
+        except ClientResponseError:
+            logger.error(f'Invalid content type received, expected json but got {response.headers["Content-Type"]}')
+        except Exception:
+            logger.exception('Unknown exception occurred while retrieving response data.')
+        return result, response.status
+
+    async def resetConnection(self):
+        await self.stop()
+        self.client = aiohttp.ClientSession(loop=self.loop)
+
+    async def getRequest(self, url, params=None, max_retry=5):
+        if not params:
+            params = {}
+
+        result = None
+        status = self.SERVICE_UNAVAILABLE_STATUS
+        retry_count = 1
+        while retry_count <= max_retry:
+            try:
+                async with self.client.get(url, params=params) as resp:
+                    result, status = await self.getJson(resp)
+
+                if result is None:
+                    logger.error(f'No result received from {url}. Retry {retry_count} of {max_retry}')
+                else:
+                    return result, status
+            except ConnectionResetError:
+                logger.error(f'Connection reset on get request - {url}. Retry {retry_count} of {max_retry}')
+            except Exception:
+                logger.exception(f'Unknown exception occurred on get request {url}. Retry {retry_count} of {max_retry}')
+
+            # Only gets here on failure
+            retry_count += 1
+            await self.resetConnection()
+            await asyncio.sleep(0.3)

```

Comment:
Something to table for future discussions but this number could potentially be reduced. We could probably get away with something smaller, seeing as how all these services are running on localhost.

Reply:
I agree. Can look into some kind of simple exponential back-off maybe